File: dotnet-src/heir-time-api/Models/Project.cs
using MongoDB.Bson.Serialization.Attributes;

namespace heir_time_api.Models;

public class Project
{
    [BsonId]
    [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("projectName")]
    public required string ProjectName { get; set; }

    [BsonElement("owner")]
    public string? Owner { get; set; }

    [BsonElement("admins")]
    public List<string> Admins { get; set; } = new List<string>();

    [BsonElement("users")]
    public List<string> Users { get; set; } = new List<string>();

    [BsonElement("items")]
    public List<Item>? Items { get; set; } = new List<Item>();

}

File: client/src/types/models/project.ts
import { Item } from './item';

export interface Project {
  id: string;
  projectName: string;
  owner: string;
  admins: string[];
  users: string[];
  items: Item[];
}

export interface NewProjectPayload {
  projectName: string;
  admins: string[];
  users: string[];
  items: Item[];
}
